packages/core/content-type-builder/admin/src/components/DataManagerProvider/tests/reducer_edit_attribute_action.test.js

```diff
@@ -1366,5 +1366,271 @@ describe('CTB | components | DataManagerProvider | reducer | EDIT_ATTRIBUTE', ()
         expect(reducer(state, action)).toEqual(expected);
       });
     });
+
+    describe('Editing a relation and preserve plugin options', () => {
```

Comment:
I'm worried this only addresses relations on the first level of the schema – not in repeatable fields, not in dynamic zones & not in components.

CC. @gu-stav thoughts?

Reply:
It should as they are all using the same redux action, basically the formModal component is in charge of adding/editing a field whether or not it's in a dz or component so the code should still work. Maybe adding a test will confirm.
The code has changed since I left so I cannot insure that it still works...

Reply:
If the PR _does_ work with comps etc. then we should still be adding tests to ensure that future changes don't impact it imo.

Reply:
Yes you should add tests, I believe there were some for components DZ and so one

Reply:
Thanks for the feedback, yes it works but i will add some tests

Reply:
Have added two tests now for  components, dynamic zones do not really make much sense to test here in my opinion, since there only the components are edited anyway.

But if you want, i can still add some tests for that

Reply:
You are right about DZ, especially since comonents allow only oneWay and manyWay so there's no opposite attribute.

Reply:
> You are right about DZ, especially since comonents allow only oneWay and manyWay so there's no opposite attribute.

Whilst this is the case currently, this won't be the case in a couple of weeks when we release a feature update to relations. Therefore, lets add tests for all cases please.

Reply:
I have added the tests for dynamic zones, however they are basically exactly the same as the component tests. I am not sure if they will fail with the new relation feature update.